File: 404.sum_of_left_leaves/sumOfLeftLeaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
    return sumofleftleaves(root);
};

//traverse the tree with a flag noting from left or right
var sumofleftleaves =function(root, isLeft){
    if(root === null) return 0;
    var sum = 0;
    if(root.left === null && root.right === null && isLeft){
        sum += root.val;
    }
    
    sumofleftleaves(root.left, true);
    sumofleftleaves(root.right, false);
    return sum;
    
}

File: 404.sum_of_left_leaves/sumOfLeftLeaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.sum = 0
        self.dfs(root)
        return self.sum
    
    def dfs(self, root):
        if not root:
            return
        if root.left:
            if not root.left.left and not root.left.right: # left leaves
                self.sum += root.left.val
            else:
                self.dfs(root.left)
        if root.right:
            self.dfs(root.right)
'''
left leaves, not left nodes
'''